static/app/views/dashboards/widgetBuilder/components/queryFilterBuilder.tsx

```diff
@@ -0,0 +1,188 @@
+import {Fragment} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import Input from 'sentry/components/input';
+import {SearchQueryBuilder} from 'sentry/components/searchQueryBuilder';
+import {Tooltip} from 'sentry/components/tooltip';
+import {IconAdd, IconDelete} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import {DisplayType} from 'sentry/views/dashboards/types';
+import {useWidgetBuilderContext} from 'sentry/views/dashboards/widgetBuilder/contexts/widgetBuilderContext';
+import {BuilderStateAction} from 'sentry/views/dashboards/widgetBuilder/hooks/useWidgetBuilderState';
+
+function WidgetBuilderQueryFilterBuilder() {
+  const {state, dispatch} = useWidgetBuilderContext();
+
+  const canAddSearchConditions =
+    state.displayType !== DisplayType.TABLE &&
+    state.displayType !== DisplayType.BIG_NUMBER;
+
+  const onAddSearchConditions = () => {
+    dispatch({
+      type: BuilderStateAction.SET_QUERY,
+      payload: state.query?.length ? [...state.query, ''] : ['', ''],
```

Comment:
:pray:
I should probably write some code in the hook to handle "empty" states so this is less jank. Possibly things that get added and removed may want to have their own dispatch handlers instead of a single "set X" action. Let's keep this for now and I can keep a note of updating this to make using it better

Reply:
This would also make it less painful and not force you to check the query length when rendering. We can keep this for now

Reply:
sounds good! i'll put in a todo here so i remember to change that once changes to the hook are made